vowpalwabbit/global_data.h

```diff
@@ -276,8 +278,7 @@ struct vw
 
   size_t length() { return (static_cast<size_t>(1)) << num_bits; };
 
-  std::vector<std::tuple<std::string, reduction_setup_fn>> reduction_stack;
-  std::vector<std::string> enabled_reductions;
+  std::unique_ptr<VW::default_reduction_stack_setup> learner_builder;

```

Comment:
The learner_builder should be a pointer to an abstract object, otherwise we won't be able to effectively override its behavior.

Reply:
This will be removed after merging with https://github.com/VowpalWabbit/vowpal_wabbit/pull/3176